docs/openapi.yml

```diff
@@ -926,6 +920,36 @@ paths:
                       type: string
         400:
           description: Bad request
+    patch:
```

Comment:
Sorry I'm coming to this late and may have missed some earlier discussions.

Just thought that this operation doesn't seem particularly RESTful... It's on the `/api/v1/templates/repositories` path, which is for a collection of `TemplateRepo` resources.  And a `PATCH` operation is normally for updating a single resource by posting a partial copy of that resource.

A bulk non-REST-ful operation of this sort may be better represented by something like `POST /api/v1/templates/repositories/_whatever` - The underscore is for denoting a non-RESTful API (there's no standard around this, it's just a convention I've seen used)

That's said, it's probably fine if we're not trying to be hardline RESTful

Reply:
That's a good point, I will do some research on a clearer API

Reply:
so looking at how Google and Facebook design batch APIs, I like Google's idea of using a homogenous `/batch/` API, such as `/api/v1/batch/templates/repositories`. 
- (Google sources: https://developers.googleblog.com/2018/03/discontinuing-support-for-json-rpc-and.html, https://developers.google.com/classroom/guides/batch)
- I don't mind where `/batch` goes in the path, but I think `/api/v1/batch/templates/repositories` is clearer and more extensible than `/api/v1/templates/repositories/batch`. 
- Facebook (https://developers.facebook.com/docs/graph-api/making-multiple-requests/) would use `batch` not in the path but as a query parameter, as in `/api/v1/templates/repositories/batch=[...]`, but I don't think this would be easy to doc. I think making a separate `/batch` route is the most user-friendly

Reply:
Re PATCH vs POST, I think we should keep this as PATCH, because we are (as you say) providing only a partial copy of the resource we're updating (template repositories)

Reply:
Not mind too much where `/batch` goes either, but in my mind when `repositories` is at the end of the path, I would expect the API payload to be a `TemplateRepo` resource, here it seems it is a `TemplateRepoSetting` resource instead.  That's also reason I was suggesting `POST` rather than `PATCH` (different resource)

Reply:
Oh I see - but TemplateRepoSetting is not really a new resource, but just my name for "a set of instructions describing how a resource currently residing on the origin server should be modified to produce a new version", which (as the RFC says) is what should be sent via a PATCH (source: https://williamdurand.fr/2014/02/14/please-do-not-patch-like-an-idiot/)

Reply:
Also this PATCH is idempotent, whereas POST is usually not

Reply:
Ok fair enough, sounds like the spec allow the PATCH to use any format to describe the changes.  My previous understanding was wrong then.